packages/core/src/Stores/contract-store.js

```diff
@@ -256,8 +257,21 @@ export default class ContractStore extends BaseStore {
             this.barriers_array = this.createBarriersArray(contract_info, is_dark_mode);
             return;
         }
+        if (
+            this.barriers_array.length === 1 &&
+            isResetContract(contract_info.contract_type) &&
+            contract_info.reset_time
+        ) {
+            this.barriers_array = this.createBarriersArray(contract_info, is_dark_mode);
```

Comment:
When the initial (default) barrier exists and we starting to receive `reset_time` from BE, then it's time to draw one more barrier - reset barrier.

Reply:
@kate-deriv shouldn't we put this condition inside `if (contract_info) {` since it's using `contract_info`?

Reply:
Will have a look